GNUmakefile

```diff
@@ -15,8 +15,20 @@ ifndef EXTRA_LDFLAGS
 EXTRA_LDFLAGS=
 endif
 
+ifndef EXTRA_LIBS
+EXTRA_LIBS=
+endif
+
+ifndef OPT_FLAGS
+OPT_FLAGS = -O2
+endif
+
+ifndef WERROR
+WERROR = -Werror
+endif
+
 CPPFLAGS = $(EXTRA_CPPFLAGS) -Isljit_src
-CFLAGS += -O2 -Wall -Wextra -Wconversion -Wsign-compare -Werror
+CFLAGS += $(OPT_FLAGS) -Wall -Wextra -Wconversion -Wsign-compare -Wdeclaration-after-statement $(WERROR)
```

Comment:
Is `-Wdeclaration-after-statement` widely used? Maybe all warnings should go to a macro as well, so people can define it in any way they want.

On the other hand, this makefile does not produce any libraries, so never used by any real project. So complicating it is not worth it. I am not sure.

Reply:
I think it is worth adding, unless we are no longer aiming for C89 compatibility anymore as it is only supported starting with C99

Reply:
Yes, we can do this change if it does not negatively impact systems where pthreds is just a simple library without any special casing. I had issues with using libraries too early and getting unresolved references in the past.